### ISSUE TITLE
bin update fails (error=binary path not found)

### ISSUE THREAD

**Issue by einarpersson** [marcosnils/bin#190]
Hi! I am new here, and I'm very excited by this little tool. I am using [Obtainium](https://github.com/ImranR98/Obtainium) on Android which is similar for that platform.

Anyway, everthing seems to work so far, _exceeept_

```
❯ bin update aichat --debug
   • debug logs enabled, version: 0.17.3
commit: 9da887da9ce476a54e25828bad962a1cae2da17c
built at: 2024-01-31T03:39:37Z
built by: goreleaser

   • Config directory is: /home/einar/config/bin
   • Download path set to $HOME/.local/bin
   ⨯ command failed            error=binary path /home/einar/aichat not found
```

I don't understand why it would think it is in `/home/einar/aichat`. It looks like a bug to be (because my config is alright, yes?)

```json
{
    "default_path": "$HOME/.local/bin",
    "bins": {
        "$HOME/.local/bin/aichat": {
            "path": "$HOME/.local/bin/aichat",
            "remote_name": "aichat",
            "version": "v0.14.0",
            "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
            "url": "https://github.com/sigoden/aichat",
            "provider": "github",
            "package_path": "aichat"
        },
        "$HOME/.local/bin/bin": {
            "path": "$HOME/.local/bin/bin",
            "remote_name": "bin",
            "version": "v0.17.3",
            "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
            "url": "github.com/marcosnils/bin",
            "provider": "github",
            "package_path": ""
        }
    }
}
```

Thanks!

**Comment by marcosnils** [comment]
Hey! thx for reporting this. Should be fixed in v0.17.4. Let me know in case it doesn't work as you're expecting. 

Cheers!

**Comment by einarpersson** [comment]
Works! And it was a great experience to update `bin` using bin  ;)
